iceoryx_utils/include/iceoryx_utils/internal/cxx/string.inl

```diff
@@ -26,13 +26,13 @@ inline constexpr string<Capacity>::string() noexcept
 template <uint64_t Capacity>
```

Comment:
The default constructor in line 22 can be replaced with =default in the header declaration